File: 0007-reverse-integer/solution.ts
function reverse(x: number): number {
  const min = -2147483648;
  const max = 2147483647;

  let answer = parseInt(Math.abs(x).toString().split('').reverse().join(''));
  if (x < 0) answer = -answer;
  if (answer < min || answer > max) answer = 0;
  return answer;
}

File: 0007-reverse-integer/solution.cpp
class Solution {
public:
    int reverse(int x) {
        string s = to_string(abs((long long)x));
        std::reverse(s.begin(), s.end());
        long long ans = stoll(s);
        ans = x < 0 ? -ans : ans;
        return (ans > INT_MAX || ans < INT_MIN) ? 0 : ans;
    }
};
